File: src/EPI_CSharp/Problem_5_9.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EPI_CSharp
{
    public static class Problem_5_9
    {
        public static bool IsPalindrome(long value)
        {
            if (value < 0)
                return false;

            var numDigits = (long)Math.Floor(Math.Log10(value)) + 1;
            var numMask = (long)Math.Pow(10, numDigits - 1);

            for (var i = 0; i < numDigits / 2; ++i)
            {
                if (value / numMask != value % 10)
                    return false;
                value %= numMask;
                value /= 10;
                numMask /= 100;
            }
            return true;
        }
    }
}

File: src/EPI/EPI_TypeScript/src/problem_5_9.ts

export function IsPalindrome(value: number): boolean {
    if (value < 0)
        return false;
    value = Math.floor(value); // remove decimal fraction 

    const numDigits = Math.floor(Math.log(value) / Math.LOG10E) + 1;
    let numMask = Math.pow(10, numDigits - 1);

    for (let i = 0; i < Math.floor(numDigits / 2); ++i) {
        if (Math.floor(value / numDigits) !== value % 10)
            return false;

        value /= numMask;
        value /= 10;
        value = Math.floor(value);

        numMask = Math.floor(numMask / 100);
    }
    return true;
}